banyand/liaison/grpc/grpc.go

```diff
@@ -35,10 +39,33 @@ type Server struct {
 	log      *logger.Logger
 	ser      *grpclib.Server
 	pipeline queue.Queue
+	repo     discovery.ServiceRepo
 }
 
-func NewServer(ctx context.Context, pipeline queue.Queue) *Server {
-	return &Server{pipeline: pipeline}
+func (s *Server) Rev(message bus.Message) (resp bus.Message) {
+	data, ok := message.Data().([]byte)
+	if !ok {
+		s.log.Warn().Msg("invalid event data type")
+		return
+	}
+	shardEvent := v1.GetRootAsShardEvent(data, 0)
```

Comment:
I am curious how can we ensure that the data received is `ShardEvent`?

Reply:
The `event.TopicShardEvent` indicates the underlying binaries are marshaled from `ShardEvent`. That's why we introduce `common.KindVersion` and `bus.Topic` into the event bus system. 

As you know, GO lacks the generic type. We choose the interface for generics instead of reflection or code-generation because GO will support generic in the next generation, GO2. The interface style will help banyandb switch to such a generic type conveniently.

Reply:
I see. I found this can be guaranteed by the `bus`. Make sense to me